diff --git a/js/jquery.social.stream.wall.1.6.js b/js/jquery.social.stream.wall.1.6.js
index faca13d..b85d2b6 100644
--- a/js/jquery.social.stream.wall.1.6.js
+++ b/js/jquery.social.stream.wall.1.6.js
@@ -29,7 +29,7 @@ jQuery(window).load(function(){
 	jQuery.each($container,function(){
 	//	sortstream(jQuery(this),'asc');
 		jQuery('li .section-thumb img, li .section-text img',jQuery(this)).css('opacity',0).show().fadeTo(800,1);
-		jQuery(this).isotope('layout');
+    jQuery(this).isotope({layoutMode: 'fitRows'});
 	});
 	
 	function sortstream(obj,d){
