diff --git a/plugins/table/views_bootstrap_table_plugin_style.inc b/plugins/table/views_bootstrap_table_plugin_style.inc
index 25817e4..701ac1a 100644
--- a/plugins/table/views_bootstrap_table_plugin_style.inc
+++ b/plugins/table/views_bootstrap_table_plugin_style.inc
@@ -32,6 +32,7 @@ class ViewsBootstrapTablePluginStyle extends views_plugin_style_table {
         'bordered' => t('Bordered'),
         'hover' => t('Hover'),
         'condensed' => t('Condensed'),
+        'responsive' => t('Responsive'),
       ),
     );
   }
diff --git a/templates/table/views-bootstrap-table-plugin-style.tpl.php b/templates/table/views-bootstrap-table-plugin-style.tpl.php
index b8ef715..bc0e3e7 100755
--- a/templates/table/views-bootstrap-table-plugin-style.tpl.php
+++ b/templates/table/views-bootstrap-table-plugin-style.tpl.php
@@ -24,6 +24,8 @@
   <h3><?php print $title ?></h3>
 <?php endif ?>
 
+<div <?php if (strpos($classes, 'table-responsive')) { print 'class="table-responsive"'; } ?>>
+
 <table <?php if ($classes) { print 'class="'. $classes . '" '; } ?><?php print $attributes; ?>>
   <?php if (!empty($title) || !empty($caption)) : ?>
     <caption><?php print $caption . $title; ?></caption>
@@ -51,3 +53,4 @@
   <?php endforeach; ?>
   </tbody>
 </table>
+</div>
